### ISSUE TITLE
Array of zip broken in `try it out` in api doc for `POST /contests/{cid}/problems` endpoint

### ISSUE THREAD

**Issue by RagnarGrootKoerkamp** [DOMjudge/domjudge#1239]
Filling in the form under the `try it out` for the `POST` `/contests/{cid}/problems` endpoint results in the following curl:

```
curl -X POST "https://dj.storm.vu/api/v4/contests/1/problems" -H "accept: application/json" -H "Authorization: Basic <pass>" -H "Content-Type: multipart/form-data" -F "zip=@greetings.zip;type=application/zip" -F "problem=1"
```
which gives this output:
```
{
  "problem_ids": [],
  "messages": []
}
```

It seems that the `-F zip=@...` sends a single zip file instead of an array of one zip, and this single zip is not found by the backend, resulting in an empty return value.

Replacing it by `-F zip[]=@...` (as done in the [import script](https://github.com/DOMjudge/domjudge/blob/main/misc-tools/import-contest.sh#L110)) fixes it.

There are 2 bugs here:
1. [upstream] Apidoc is wrong and should send an array of zips, instead of multiple single-zips under the same key.
2. [convenience] The domjudge backend could still accept a single non-array zip under key `zip`.

------

Ah, I finally found the solution for my python client: Instead of passing the file as
```
        files=[('zip', zipfile)],
```
I have to explicitly make the key `zip[]` as well:
```
        files=[('zip[]', zipfile)],
```

So basically, it seems that the key here is just `zip[]` instead of `zip`. I don't know whether this is common practice or not for 'array of file' fields, but documentation could be better. This is not mentioned in the apidoc (which also does it wrong), nor in the [CCS spec](https://ccs-specs.icpc.io/contest_api) (it appears that `POST /contests/{cid}/problems` isn't even part of that).

**Comment by meisterT** [comment]
I first fixed the API doc to reflect reality.

Uploading multiple problems in one call may seem convenient (and that's why I did it back then) but it is also error prone since you easily run into all sorts of limits. If we touch this we might as well decide to only accept one problem zip at a time.

**Comment by nickygerritsen** [comment]
We stopped using `zip[]` on `main` so from 8.x onwards it will just be `zip`, making this issue void.
